docs/source/content/supervised/knn.rst

```diff
@@ -2,32 +2,160 @@
+infinity all unidentified data points will always be classified to one class or the other
+depending on which class has more data points. You don't want this to happen,
+so it is wise to choose a k that is relatively small.
+
+Conclusion
+------------
+
+Here are some things to take away:
+
+- The different methods to KNN only affect the performance, not the output
+- The Brute Method is best when the dimensions of the points or the number of points are small
+- The K-D Tree Method is best when you have a larger data set
+- SKLearn KNN classifier has a auto method which decides what method to use given what data it's trained on. 
+
+Choosing the value of k will drastically change how the data is classified. A higher k value will ignore outliers to the data 
+and a lower will give more weight to them. If the k value is too high it will not be able to classify the data, so k needs to 
+be relatively small. 
+
+Motivation
+------------
+
+So why would someone use this classifier over another? Is this the best classifier? The answer to these questions are that it depends. 
+There is no classifier that is best, it all depends on the data that a classifier is given. KNN might be the best for one dataset but 
+not another. It's good to know about other classifiers like Support Vector Machines, and then decide which one best classifies the 
+a given dataset.
+
+Code Example
+-------------
+
+Check out our code, `knn.py`_ to learn how to implement a k nearest neighbor classifier using Python's Scikit-learn library. 
+More information about Scikit-Learn can be found `here`_. 
+
+`knn.py`_, Classifies a set of data on breast cancer, loaded from Scikit-Learn's dataset library. 
+The program will take the data and plot them on a graph, then use the KNN algorithm to best separate the data. 
+The output should look like this:
+
+.. figure:: _img/knn_output_k9.png
+   :scale: 50%
+   :alt: KNN k = 9 output
+
+The green points are classified as benign.
+The red points are classified as malignant.
+The boundary line is the prediction that the classifier makes. This boundary line is determined by the k value, for this instance
+k = 9. 
+
+This loads the data from the Scikit-Learn's dataset library. You can change the data to whatever you would like. 
+Just make sure you have, data points and an array of targets to classify those data points. 
+
+.. code:: python
+
+    dataCancer = load_breast_cancer()
+    data = dataCancer.data[:, :2]
+    target = dataCancer.target
+
+You can also change the k value or n_neighbors value that will change the algorithm. It is suggested that you 
+choose a k that is relatively small. You can also change the algorithm used, the options are 
+{‘auto’, ‘ball_tree’, ‘kd_tree’, ‘brute’}. These don't change the output of the prediction they just will 
+change the time it takes to predict the data. You can change it here in the code, try changing the value of n_neighbors to 1. 
```

Comment:
changing n_neighbors is not what is being discussed in this paragraph, I suggest making a new paragraph to explicitly discuss changing the neighbors.

Reply:
Actually, it goes back and forth.  Definitely separate the algorithm and the k into two separate paragraphs.